bool initialized_
int16 data